src/Dialect/ONNX/ONNXOps/Math/DFT.cpp

```diff
@@ -24,34 +25,43 @@ using namespace onnx_mlir;
 
 namespace onnx_mlir {
 
-template <>
-LogicalResult ONNXDFTOpShapeHelper::computeShape() {
-  ONNXDFTOpAdaptor operandAdaptor(operands, op->getAttrDictionary());
+template <typename OP_TYPE>
+LogicalResult ONNXGenericDFTOpShapeHelper<OP_TYPE>::customComputeShape(
+    IndexExpr &axis) {
+  typename OP_TYPE::Adaptor operandAdaptor(operands, op->getAttrDictionary());
+
   // Get info about input data operand.
   Value input = operandAdaptor.getInput();
   // Get the rank to compensate for N dimensions.
   int64_t rank = createIE->getShapedTypeRank(input);
 
-  // Axis is a required attribute and should have a default value of 1.
-  int64_t axis = operandAdaptor.getAxis();
+  // Check if the dimension for axis is a literal and in range.
+  if (!axis.isLiteral())
+    return op->emitError("Can not perform Discrete Fourier Transform on "
+                         "dynamic dimensions at this time");
 
   // OneSided is a required attribute and should have default value of 0.
   // However oneSided can also be a value of 1 and if so a specific shape is
-  // expected Values can be 0 or 1.
+  // expected Values can be 0 or 1. When onesided is 0 it is complex input and
+  // when onesided is 1 it is a real input.
   int64_t oneSided = operandAdaptor.getOnesided();
+  int64_t axisValue = axis.getLiteral();
+
   bool isOneSided = (oneSided == 0);
+  bool isAxis = (axisValue == 0);
 
   // Compute outputDims for DFT.
+
   LiteralIndexExpr one(1);
   DimsExpr outputDims;
-  for (int64_t i = 0; i < rank; ++i) {
-    if (isOneSided) {
+  for (int64_t i = 0; i < rank - 1; ++i) {
+    if (isOneSided) { // onesided is 0
```

Comment:
Does the code and the comments match?  To me, `if(isOneSided)` means isOneSide is 1.

Reply:
Yes the code and comment does match.... Look at line 51. IsOneSided means oneSided == 0 but I can switch the ordering if that would make more sense to you.